aopy/analysis/controllers.py

```diff
@@ -0,0 +1,204 @@
+# controllers.py
+# 
+# Code relating to spectral analysis of feedforward & feedback controllers, based on the paper:
+#
+# Yamagami M, Peterson LN, Howell D, Roth E, Burden SA. Effect of Handedness on Learned Controllers 
+# and Sensorimotor Noise During Trajectory-Tracking. IEEE Trans Cybern. 2023 Apr;53(4):2039-2050. 
+# doi: 10.1109/TCYB.2021.3110187. Epub 2023 Mar 16. https://pubmed.ncbi.nlm.nih.gov/34587106/
+
+import numpy as np
+import math as m
+from .base import calc_freq_domain_values
+
+def get_machine_dynamics(freqs, exp_freqs, order):
+    '''
+    Returns the machine dynamics at each experimental frequency, given the system order. 
+
+    Args:
+        freqs (nt/2,): array of non-negative frequencies (essentially the x axis of a spectrogram)
+        exp_freqs (nfreq,): list or array of frequencies used to generate experimental signals (reference and/or disturbance)
+        order (int): the order of the system (e.g. 0 for position control, 1 for velocity control, 2 for acceleration control)
+
+    Returns:
+        M (nfreq,): array of machine dynamics at each experimental frequency
+
+    '''
+    s = 1.j*2*m.pi*freqs
+    exp_idx = np.isin(freqs.round(4), exp_freqs)
+
+    if order == 0:
+        M = 1./np.ones((len(s),))
+    elif order == 1:
+        M = 1./(s)
+    elif order == 2:
+        M = 1./(s**2+s)
+    else:
+        print('Order not recognized!')
+
+    return M[exp_idx]
+
```

Comment:
nice, this looks good.

Reply:
This only calculates the M at the stimulated frequencies right? Might be a bit harder to use this function if you're doing other analysis with signals at a different frequency (e.g. non-stimulus frequencies).

Reply:
I changed exp_freqs to be an optional argument so that M at all frequencies is returned by default